lib/mruby/build/command.rb

```diff
@@ -126,6 +127,31 @@ def define_rules(build_dir, source_dir='')
       end
     end
 
+    def compiles?(source_text)
+      infile = Tempfile.new ['', '.c']
+      infile.write source_text
+      infile.close
+      cwd = Dir.pwd
+      is_success = false
+      # Change to a tmp dir when compiling so we don't litter compiler artifacts
+      Dir.mktmpdir do |tmpdir|
+        Dir.chdir tmpdir
+        sh(command, infile.path, verbose: false) { |retval, _| is_success = retval }
```

Comment:
There is a bit of a nuisance here. You cannot silence stderr with the Rake `sh` method. When testing for features it may be a legitimate failure, which is why we're asking the compiler. Unfortunately gcc and clang output the compiler errors to `stderr`. Doing a shell redirection isn't cross platform. We could use `Open3.capture3` instead of `sh`, I didn't know if it is okay to require and use.